### PR TITLE
CMake: Handle discovering libusb without pkg-config

### LINKED ISSUE #1027: libusb discovery fails on FreeBSD
In 0.25 you've added pkgconfig discovery for libusb and for some reason the code fails to discover libusb.

On FreeBSD libusb comes from the base and isn't discoverable by pkgconfig.

The alternative code path fails for some reason.

### PR DESCRIPTION
**Pull request by pcercuei**
Don't require pkg-config for libusb discovery. If pkg-config is not found, use the standard find_library / find_path approach.

Fixes #1027.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -213,14 +213,14 @@ if (WITH_USB_BACKEND)
...
-		else()
-			set(LIBUSB_LIBRARIES ${LIBUSB_LINK_LIBRARIES})
-			set(LIBUSB_INCLUDE_DIR ${LIBUSB_INCLUDE_DIRS})
-		endif()
+	endif()
+	if (LIBUSB_LINK_LIBRARIES AND LIBUSB_INCLUDE_DIRS)
+		set(LIBUSB_LIBRARIES ${LIBUSB_LINK_LIBRARIES})
+		set(LIBUSB_INCLUDE_DIR ${LIBUSB_INCLUDE_DIRS})
+	else()
+		#Handle FreeBSD libusb and Linux libusb-1.0 libraries
```
Comment (nunojsa) [inline CMakeLists.txt:221]: nit: maybe add a space to be consistent with the rest of the file...

**Review by nunojsa — APPROVED** [review]

**Comment by rgetz** [comment]
Works for me.
```
rgetz@qemu:~/libiio/build $ freebsd-version 
13.2-RELEASE
rgetz@qemu:~/libiio/build $ cmake ../
...
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "1.8.1") 
-- Checking for module 'libusb-1.0'
--   Found libusb-1.0, version 1.0.13
-- Looking for libusb_get_version
-- Looking for libusb_get_version - found
...
-- Features enabled : xml zstd network dns-sd avahi ipv6 usb utils man-utils
-- Features disabled: bonjour serial local local-dmabuf local-mmap hwmon examples iiod modules usb-dynamic network-dynamic serial-dynamic udev-rule doc man python-bindings c#-bindings c++-bindings
rgetz@qemu:~/libiio/build $ make
rgetz@qemu:~/libiio/build $ ./tests/iio_info --version
iio_info version: 1.0 (git tag:v1.0)
Libiio version: 1.0 (git tag: v1.0) backends: xml ip usb
rgetz@qemu:~/libiio/build $ sudo ./tests/iio_info -S
Available contexts:
	0: 0456:b673 (Analog Devices Inc. PlutoSDR (ADALM-PLUTO)), serial=10447376de0b000213002900fafa9908ec [usb:3.3.5]
```

and the default, without PkgConfig works too...

This seems to fix the freebsd / usb configure/build issue - but still needs the devd rule to install properly in freebsd...

```
rgetz@qemu:~/libiio/build $ git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce0763ec..e257a28b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -657,6 +657,7 @@ toggle_iio_feature("${WITH_USB_BACKEND_DYNAMIC}" usb-dynamic)
 toggle_iio_feature("${WITH_NETWORK_BACKEND_DYNAMIC}" network-dynamic)
 toggle_iio_feature("${WITH_SERIAL_BACKEND_DYNAMIC}" serial-dynamic)
 toggle_iio_feature("${INSTALL_UDEV_RULE}" udev-rule)
+toggle_iio_feature("${INSTALL_DEVD_RULE}" devd-rule)
 #add iiod settings
 list(APPEND IIO_FEATURES_ON ${IIOD_FEATURES_ON})
 list(APPEND IIO_FEATURES_OFF ${IIOD_FEATURES_OFF})
diff --git a/cmake/Install.cmake b/cmake/Install.cmake
index 41263b02..87660f7a 100644
--- a/cmake/Install.cmake
+++ b/cmake/Install.cmake
@@ -186,6 +186,17 @@ if (WITH_USB_BACKEND AND CMAKE_SYSTEM_NAME MATCHES "^Linux")
        endif()
 endif()
 
+if (WITH_USB_BACKEND AND CMAKE_SYSTEM_NAME MATCHES "^FreeBSD")
+       option(INSTALL_DEVD_RULE "Install a devd rule for detection of USB devices" ON)
+
+       if (INSTALL_DEVD_RULE)
+               set(DEVD_RULES_INSTALL_DIR /etc/devd CACHE PATH "default install path for udev rules")
+
+               configure_file(libiio.devd.cmakein ${CMAKE_CURRENT_BINARY_DIR}/libiio.conf @ONLY)
+               install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libiio.conf DESTINATION ${DEVD_RULES_INSTALL_DIR})
+       endif()
+endif()
+
 if (WITH_NETWORK_BACKEND_DYNAMIC)
        install(TARGETS iio-ip LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/libiio)
 endif()


rgetz@qemu:~/libiio/build $ cat ../libiio.devd.cmakein 
# libiio devd rule for devd
# $FreeBSD$
#

notify 0 {
        match "system"          "USB";
        match "subsystem"       "DEVICE";
        match "type"            "ATTACH";
        action "@CMAKE_INSTALL_FULL_BINDIR@/iio_info -S usb=$vendor:$product && chmod 666 /dev/$cdev";
};
```

**Review by rgetz — APPROVED** [review]
looks good - your call on including the devd rule.

**Comment by pcercuei** [comment]
I'll include the devd rule in a subsequent PR.

**Pull request merged**